pallets/asset-registry/src/lib.rs

```diff
@@ -163,83 +194,134 @@ pub mod pallet {
 		StorageMap<_, Blake2_128Concat, T::AssetNativeLocation, T::AssetId, OptionQuery>;
 
 	#[pallet::storage]
-	#[pallet::getter(fn asset_metadata)]
-	/// Metadata of an asset.
-	pub type AssetMetadataMap<T: Config> =
-		StorageMap<_, Twox64Concat, T::AssetId, AssetMetadata<BoundedVec<u8, T::StringLimit>>, OptionQuery>;
+	/// Number of accounts that paid existential deposits for insufficient assets.
+	/// This storage is used by `SufficiencyCheck`.
+	pub type ExistentialDepositCounter<T: Config> = StorageValue<_, u128, ValueQuery>;
 
 	#[allow(clippy::type_complexity)]
 	#[pallet::genesis_config]
 	pub struct GenesisConfig<T: Config> {
-		pub registered_assets: Vec<(Vec<u8>, T::Balance, Option<T::AssetId>)>,
+		//asset_id, name, existential deposit, symbol, decimals, xcm_rate_limit, is_sufficient
+		pub registered_assets: Vec<(
+			Option<T::AssetId>,
+			Option<Vec<u8>>,
+			Balance,
+			Option<Vec<u8>>,
+			Option<u8>,
+			Option<Balance>,
+			bool,
+		)>,
 		pub native_asset_name: Vec<u8>,
-		pub native_existential_deposit: T::Balance,
+		pub native_existential_deposit: Balance,
+		pub native_symbol: Vec<u8>,
+		pub native_decimals: u8,
 	}
 
 	#[cfg(feature = "std")]
 	impl<T: Config> Default for GenesisConfig<T> {
 		fn default() -> Self {
 			GenesisConfig::<T> {
 				registered_assets: vec![],
-				native_asset_name: b"BSX".to_vec(),
-				native_existential_deposit: Default::default(),
+				native_asset_name: b"HDX".to_vec(),
+				native_existential_deposit: DEFAULT_ED,
+				native_symbol: b"HDX".to_vec(),
+				native_decimals: 12,
 			}
 		}
 	}
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
-			// Register native asset first
-			// It is to make sure that native is registered as any other asset
-			let native_asset_name = Pallet::<T>::to_bounded_name(self.native_asset_name.to_vec())
-				.map_err(|_| panic!("Invalid native asset name!"))
-				.unwrap();
-
-			AssetIds::<T>::insert(&native_asset_name, T::NativeAssetId::get());
-			let details = AssetDetails {
-				name: native_asset_name,
-				asset_type: AssetType::Token,
-				existential_deposit: self.native_existential_deposit,
-
-				xcm_rate_limit: None,
-			};
-
-			Assets::<T>::insert(T::NativeAssetId::get(), details);
-
-			self.registered_assets.iter().for_each(|(name, ed, id)| {
-				let bounded_name = Pallet::<T>::to_bounded_name(name.to_vec())
-					.map_err(|_| panic!("Invalid asset name!"))
-					.unwrap();
-				let _ = Pallet::<T>::register_asset(bounded_name, AssetType::Token, *ed, *id, None)
-					.map_err(|_| panic!("Failed to register asset"));
-			})
+			let _ = with_transaction(|| {
```

Comment:
i wonder why you ignore the error.

Reply:
I don't think this can fail. If something inside of this  block fail, it will panic on `expect` or `unwrap`.